src/settings.rs

```diff
@@ -6,7 +6,11 @@ use crate::{
 };
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
-
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq, Eq, Copy)]
+#[serde(rename_all = "camelCase")]
+pub struct PaginationSetting {
+    pub max_total_hits: u32
```

Comment:
I checked with the engineers working on Meilisearch. `usize` is better as it adapts do you architecture.

```suggestion
    pub max_total_hits: usize
```